readthedocs/projects/models.py

```diff
@@ -144,11 +144,18 @@ class AddonsConfig(TimeStampedModel):
     Everything is enabled by default.
     """
 
-    DOC_DIFF_DEFAULT_ROOT_SELECTOR = "[role=main]"
-    LINKPREVIEWS_DEFAULT_ROOT_SELECTOR = "[role=main] a.internal"
-    LINKPREVIEWS_DOCTOOL_NAME_CHOICES = (
-        ("sphinx", "Sphinx"),
-        ("other", "Other"),
+    DEFAULT_ROOT_SELECTOR = ", ".join(
+        [
+            "[role=main]",
+            "main",
+            "article",  # Pelican, Docusaurus, Antora, Jekyll
+            "div#content",  # AsciiDoctor
+            "main > div > div.md-content",  # Material for MkDocs
+            "article#main",  # Docsify
+            "div.body",
+            "div.document",
+            "h1",  # Very generic fallback
+        ]
```

Comment:
This should fallback to select the whole document (maybe `body`, but it can be cases where a doc doesn't have a body, but that's probably an edge case, and probably invalid document)